crypto/secp256k1/secp256k1.go

```diff
@@ -118,10 +118,7 @@ func (privKey PrivKey) Sign(msg []byte) ([]byte, error) {
 	priv, _ := secp256k1.PrivKeyFromBytes(privKey)
 
 	sum := sha256.Sum256(msg)
-	sig, err := ecdsa.SignCompact(priv, sum[:], false)
-	if err != nil {
-		return nil, err
-	}
+	sig := ecdsa.SignCompact(priv, sum[:], false)
 
 	// remove the first byte which is compactSigRecoveryCode
 	return sig[1:], nil
```

Comment:
This will panic if `sig` is `nil`. Should we check for `nil` value?

Reply:
The call to [`ecdsa.SignCompact`](https://github.com/btcsuite/btcd/blob/btcec/v2.3.4/btcec/ecdsa/signature.go#L238) is then handed over to `github.com/decred/dcrd/dcrec/secp256k1/v4` module's [`ecdsa.SignCompact`](https://github.com/decred/dcrd/blob/dcrec/secp256k1/v4.0.1/dcrec/secp256k1/ecdsa/signature.go#L712). The function there always seems to return a value. I am not seeing where a `nil` could come from.

The change that the `btcec/v2` project made (btcsuite/btcd@e5d15fddb9c486dfe382bc95dc38d2fd247813bf) was replace:
```go
return secp_ecdsa.SignCompact(key, hash, isCompressedKey), nil
```
with
```go
return secp_ecdsa.SignCompact(key, hash, isCompressedKey)
```

Previously `nil` was not being checked in CometBFT, so I think there is no need to check for `nil` now. 🤔